### PR TITLE
feat(snapshot): plugin changes for setvolumeattributes

### PR DESCRIPTION
**Pull request by hrudaya21**

### CODE REVIEW

**Review by abhilashshetty04 — APPROVED** [review]

File: control-plane/plugin/src/resources/volume.rs [hunk 1]
```diff
@@ -225,6 +227,63 @@ impl Scale for Volume {
...
+    ) -> PluginResult {
+        let parts: Vec<&str> = property.trim().split('=').collect();
+        if parts.len() != 2 {
+            return Err(Error::SetVolumePropertyParseError {
+                reason: format!("volume: {id}, invalid property format: {property}, enter in the format property=value"),
+            });
+        }
+        let left = parts[0].trim();
+        let right = parts[1].trim();
+        let property_body = match left {
```
Comment (Abhinandan-Purkait) [inline control-plane/plugin/src/resources/volume.rs]: Unrelated to this PR. @tiagolobocastro Just wondering if we should use strum on generated code to get these conversions?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/plugin/src/resources/mod.rs [hunk 2]
```diff
@@ -69,6 +70,18 @@ pub enum ScaleResources {
...
 
+/// The types of resources that support the 'SetProperty' operation.
+#[derive(clap::Subcommand, Debug)]
+pub enum SetPropertyResources {
+    /// Set volume property.
+    Volume {
+        /// ID of the volume.
+        id: VolumeId,
+        /// Property of volume.
+        property: String,
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/mod.rs]: let's use explicit types here too

File: control-plane/plugin/src/resources/volume.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/plugin/src/resources/volume.rs]: I think we should use explicit types to begin with, perhaps even use the openapi properties in the CLI

File: control-plane/plugin/src/resources/mod.rs (hunk 2 above)
Comment (hrudaya21) [reply control-plane/plugin/src/resources/mod.rs]: Use explicit field with clap trait now.

File: control-plane/plugin/src/resources/mod.rs [hunk 3]
```diff
@@ -69,6 +70,19 @@ pub enum ScaleResources {
...
+/// The types of resources that support the 'SetProperty' operation.
+#[derive(clap::Subcommand, Debug)]
+pub enum SetPropertyResources {
+    /// Set volume property.
+    Volume {
+        /// ID of the volume.
+        id: VolumeId,
+        /// Max snapshot limit per volume.
+        #[clap(long)]
+        max_snapshots: u32,
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/mod.rs]: Shouldn't this be an enum?

Comment (Abhinandan-Purkait) [reply control-plane/plugin/src/resources/mod.rs]: Yeah, was thinking the same, should it be rather an enum of Vol Properties then we can add more?

Comment (tiagolobocastro) [reply control-plane/plugin/src/resources/mod.rs]: https://github.com/openebs/mayastor-control-plane/tree/set-properties

**Comment by hrudaya21** [comment]
bors try

File: control-plane/plugin/src/resources/error.rs [hunk 4]
```diff
@@ -81,8 +81,8 @@ pub enum Error {
         source: openapi::tower::client::Error<openapi::models::RestJsonError>,
     },
     /// Error when set volume property request fails.
-    #[snafu(display("Failed to set volume {id} property, Error {source}"))]
-    ScaleVolumePropertyError {
+    #[snafu(display("Failed to set volume {id} property, Error: {source}"))]
```
Comment (tiagolobocastro) [inline control-plane/plugin/src/resources/error.rs]: Existing ones don't have ':'.
Let's keep it consistent

**Review by tiagolobocastro — APPROVED** [review]

**Comment by hrudaya21** [comment]
bors merge

**Pull request merged**